docs/source/sizeof-plugins.rst

```diff
@@ -0,0 +1,42 @@
+.. _order:
+
+``sizeof`` Plugins
+========
+
+.. note::
+
+   This is an advanced topic that most users won't need to worry about.
+
+When Dask needs to compute the memory footprint of an object, e.g. to determine which objects to spill to disk, it uses the `sizeof` API. Libraries which need to define a `sizeof` implementation for their own objects can use the `sizeof.register` API:
```

Comment:
I'm not sure if "memory footprint" is an English-specific phrase. We might modify things slightly, e.g. "size of an object in bytes", to make this point more clear. 

```suggestion
When Dask needs to determine the size of an object in bytes, e.g. to determine which objects to
spill to disk, it uses the ``dask.sizeof.sizeof`` registration mechanism. Users who need to
define a custom ``sizeof`` implementation for their own objects can use ``sizeof.register``:
```